### PR TITLE
Update votes with mongo objects 

### PR DESCRIPTION
**Pull request by shalithasuranga**
Fix for #44

### CODE REVIEW

File: fact-bounty-server/controllers/story.js [hunk 1]
```diff
@@ -24,35 +24,41 @@ exports.storyRange = function (req, res, next) {
...
+	}, function (mon_err, new_doc) {
+		return mon_err ? res.status(200).json(mon_err) : res.status(200).json(new_doc)
 	})
 }
 
 // @desc Change downvote count of a story
 // @access Public
 exports.storyChangeDownvoteCount = function (req, res, next) {
-	Story.updateOne({ _id: req.body.story_id }, {
+	Story.findOneAndUpdate({ _id: req.body.story_id }, {
```
Comment (Anmolbansal1) [inline fact-bounty-server/controllers/story.js]: `updateOne` method is better then `findOneAndUpdate` so there is no need to change it

File: fact-bounty-server/controllers/story.js [hunk 2]
```diff
@@ -24,35 +24,41 @@ exports.storyRange = function (req, res, next) {
...
 		$inc: {
 			fake_count: req.body.change_val
 		}
-	}, function (mon_err, affected, mon_res) {
-		return mon_err ? res.status(200).json(mon_err) : res.status(200).send()
+	}, {
+		new : true
+	}, function (mon_err, new_doc) {
+		return mon_err ? res.status(200).json(mon_err) : res.status(200).json(new_doc)
```
Comment (Anmolbansal1) [inline fact-bounty-server/controllers/story.js]: In case of error, response code shouldn't be 200

File: fact-bounty-server/controllers/story.js [hunk 3]
```diff
@@ -24,35 +24,41 @@ exports.storyRange = function (req, res, next) {
...
-	Story.updateOne({ _id: req.body.story_id }, {
+	Story.findOneAndUpdate({ _id: req.body.story_id }, {
 		$inc: {
 			mixedvote_count: req.body.change_val
 		}
-	}, function (mon_err, affected, mon_res) {
-		return mon_err ? res.status(200).json(mon_err) : res.status(200).send()
+	}, {
+		new : true
+	}, function (mon_err, new_doc) {
```
Comment (root5533) [inline fact-bounty-server/controllers/story.js]: Let's change `new_doc` variable to `updated_story` or `story`. new_doc does not suit here IMO.

Comment (shalithasuranga) [reply fact-bounty-server/controllers/story.js]: Yeah agree `update_story` looks better

**Comment by ivantha** [comment]
@root5533 Do we need any part of this now?

**Comment by root5533** [comment]
Since we are moving from mongo to flask, this will not be required. Good attempt @shalithasuranga

**Pull request closed without merging**